lib/chef/knife/core/windows_bootstrap_context.rb

```diff
@@ -286,15 +270,15 @@ def local_download_path
         # Build a URL to query www.chef.io that will redirect to the correct
         # Chef Infra msi download.
         def msi_url(machine_os = nil, machine_arch = nil, download_context = nil)
-          if config[:msi_url].nil? || config[:msi_url].empty?
+          if @config[:msi_url].nil? || @config[:msi_url].empty?
             url = "https://www.chef.io/chef/download?p=windows"
             url += "&pv=#{machine_os}" unless machine_os.nil?
             url += "&m=#{machine_arch}" unless machine_arch.nil?
             url += "&DownloadContext=#{download_context}" unless download_context.nil?
-            url += "&channel=#{config[:channel]}"
+            url += "&channel=#{@config[:channel]}"
             url += "&v=#{version_to_install}"
           else
-            config[:msi_url]
+            @config[:msi_url]
```

Comment:
that should be left without the ivar reference

Reply:
Thanks @lamont-granquist Updated.